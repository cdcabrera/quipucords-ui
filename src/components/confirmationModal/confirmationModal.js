import React from 'react';
import PropTypes from 'prop-types';
import { MessageDialog, Icon } from 'patternfly-react';
import { connect, store, reduxTypes } from '../../redux';
import helpers from '../../common/helpers';

const ConfirmationModal = ({
  show,
  title,
  heading,
  body,
  icon,
  confirmButtonText,
  cancelButtonText,
  onConfirm,
  onCancel
}) => {
  const cancel = () => {
    if (onCancel) {
      onCancel();
    } else {
      store.dispatch({
        type: reduxTypes.confirmationModal.CONFIRMATION_MODAL_HIDE
      });
    }
  };

  return (
    <MessageDialog
      primaryAction={onConfirm}
      secondaryAction={cancel}
      onHide={cancel}
      icon={icon}
      show={show}
      title={title}
      primaryActionButtonContent={confirmButtonText}
      secondaryActionButtonContent={cancelButtonText}
      primaryContent={<p>{heading}</p>}
      secondaryContent={<p>{body}</p>}
    />
  );
};

ConfirmationModal.propTypes = {
  show: PropTypes.bool.isRequired,
  title: PropTypes.string,
  heading: PropTypes.node,
  icon: PropTypes.node,
  body: PropTypes.node,
  confirmButtonText: PropTypes.string,
  cancelButtonText: PropTypes.string,
  onConfirm: PropTypes.func,
  onCancel: PropTypes.func
};

ConfirmationModal.defaultProps = {
  title: 'Confirm',
  heading: null,
  body: null,
  icon: <Icon type="pf" name="warning-triangle-o" />,
  confirmButtonText: 'Confirm',
  cancelButtonText: '',
  onConfirm: helpers.noop,
  onCancel: null
};

const mapStateToProps = state => ({ ...state.confirmationModal });

const ConnectedConfirmationModal = connect(mapStateToProps)(ConfirmationModal);

export { ConnectedConfirmationModal as default, ConnectedConfirmationModal, ConfirmationModal };
