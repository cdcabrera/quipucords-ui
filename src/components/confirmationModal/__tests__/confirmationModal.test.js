import React from 'react';
import configureMockStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import { ConnectedConfirmationModal, ConfirmationModal } from '../confirmationModal';

describe('Confirmation Modal Component', () => {
  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a connected component', () => {
    const store = generateEmptyStore({
      confirmationModal: {
        show: true,
        title: 'Confirm',
        heading: 'test',
        icon: null,
        body: 'Test body',
        confirmButtonText: 'Confirm',
        cancelButtonText: 'Cancel'
      }
    });
    const component = shallow(<ConnectedConfirmationModal />, { context: { store } });

    expect(component.dive()).toMatchSnapshot('connected');
  });

  it('should display a confirmation modal', () => {
    const onCancel = jest.fn();
    const props = {
      show: true,
      title: 'Confirm',
      heading: 'test',
      icon: null,
      body: 'Test body',
      confirmButtonText: 'Confirm',
      cancelButtonText: 'Cancel',
      onCancel
    };

    const component = mount(<ConfirmationModal {...props} />);

    expect(component.render()).toMatchSnapshot('show');

    component.find('button[className="btn btn-default"]').simulate('click');
    expect(onCancel).toHaveBeenCalledTimes(1);
  });

  it('should NOT display a confirmation modal', () => {
    const props = {
      show: false
    };

    const component = mount(<ConfirmationModal {...props} />);

    expect(component.render()).toMatchSnapshot('hidden');
  });
});
