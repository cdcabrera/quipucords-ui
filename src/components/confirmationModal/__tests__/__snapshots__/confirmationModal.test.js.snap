// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation Modal Component should NOT display a confirmation modal: hidden 1`] = `null`;

exports[`Confirmation Modal Component should display a confirmation modal: show 1`] = `
<div
  role="dialog"
>
  <div
    class="fade modal-backdrop in"
  />
  <div
    aria-describedby=""
    aria-labelledby=""
    aria-modal="true"
    class="fade message-dialog-pf in modal"
    role="dialog"
    style="display: block; padding-right: 0px;"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
        role="document"
      >
        <div
          class="modal-header"
        >
          <button
            class="close"
          >
            <span
              aria-hidden="true"
              class="pficon pficon-close"
              title="Close"
            />
            <span
              class="sr-only"
            >
              Close
            </span>
          </button>
          <h4
            class="modal-title"
            id=""
          >
            Confirm
          </h4>
        </div>
        <div
          class="modal-body"
        >
          <div
            id=""
          >
            <p>
              test
            </p>
            <p>
              Test body
            </p>
          </div>
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-default"
            type="button"
          >
            Cancel
          </button>
          <button
            class="btn btn-primary"
            type="button"
          >
            Confirm
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Confirmation Modal Component should render a connected component: connected 1`] = `
<MessageDialog
  accessibleDescription=""
  accessibleName=""
  className=""
  enforceFocus={true}
  footer={null}
  icon={null}
  onHide={[Function]}
  primaryAction={[Function]}
  primaryActionButtonBsStyle="primary"
  primaryActionButtonContent="Confirm"
  primaryContent={
    <p>
      test
    </p>
  }
  secondaryAction={[Function]}
  secondaryActionButtonBsStyle="default"
  secondaryActionButtonContent="Cancel"
  secondaryContent={
    <p>
      Test body
    </p>
  }
  show={true}
  title="Confirm"
/>
`;
